cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -3181,9 +3183,24 @@ pMaxTransactionSize =
       , Opt.help "Maximum transaction size."
       ]
 
+wordReader :: (Integral a, Bits a) => String -> ReadM a
+wordReader typeString =
+  Opt.maybeReader parser
+ where
+  -- We use 'readMaybe' instead of 'read', because the latter wouldn't output a nice error message
+  parser s =
+    case readMaybe s >>= naturalToWordMaybe >>= toIntegralSized of
+      Nothing ->
+        error $ "Cannot parse " <> s <> " as a " <> typeString
```

Comment:
Why `error` and not `fail`?

Reply:
Because I didn't knew about `fail`.. Changed :+1: